code_challenges.rb

```diff
@@ -1,49 +1,102 @@
 # ASSESSMENT 4: Ruby Coding Practical Questions
 
 # --------------------1) Create a method that takes in an array and returns an array with only odd numbers sorted from least to greatest.
+# fullArr1 = [4, 9, 0, '7', 8, true, 'hey', 7, 199, -9, false, 'hola']
 
+# def oddOnly (arr)
+# newarr2=[]
 
+# newarr = {|obj| (obj,is_a? String)|| (obj.is_a? TrueClass) || (obj.is_a? FalseClass) ||(obj.is_a? NilClass)}
 
+# newarr.each { |value|
 
+#   if (value % 2 == 1 || value % 2 == -1)
+   
+#     newarr2.push(value)
+#   end
+# }
 
-fullArr1 = [4, 9, 0, '7', 8, true, 'hey', 7, 199, -9, false, 'hola']
-# Expected output: [-9, 7, 9, 199]
-fullArr2 = ['hello', 7, 23, -823, false, 78, nil, '67', 6, 'Number']
-# Expected output: [-823, 7, 23]
-
-
-# --------------------2) Create a class called Bike that is initialized with a model, wheels, and a frame size. The default number of wheels is 2. Create a get_info method that returns a sentence with all the data from the bike object.
+# newarr2.sort
+# end
```

Comment:
this code does not run as it's written. it looks like your logic is there but there are syntax errors that are making this code break.